odoo/orm/models.py

```diff
@@ -5157,8 +5157,40 @@ def exists(self) -> Self:
             return self
         query = Query(self.env, self._table, self._table_sql)
         query.add_where(SQL("%s IN %s", SQL.identifier(self._table, 'id'), tuple(ids)))
-        self.env.cr.execute(query.select())
-        valid_ids = set([r[0] for r in self._cr.fetchall()] + new_ids)
+        real_ids = (id_ for [id_] in self.env.execute_query(query.select()))
+        valid_ids = {*real_ids, *new_ids}
+        return self.browse(i for i in self._ids if i in valid_ids)
+
+    @api.private
+    def lock_for_update(self, *, allow_referencing: bool = False) -> Self:
```

Comment:
It either locks all records or raises an exception.  The return value is pointless.
```suggestion
    def lock_for_update(self, *, allow_referencing: bool = False) -> None:
```